File: JsonSchemaVb/Student.vb
Imports System
Imports System.ComponentModel
Imports System.ComponentModel.DataAnnotations

Imports Newtonsoft.Json
Imports Newtonsoft.Json.Converters

Public Class Student

#Region "Properties"

    <Required>
    <RegularExpression("^[A-Z][0-9]{7}?$")>
    <JsonProperty(Order:=1, Required:=Required.Always)>
    Public Property Id As String

    <Required>
    <StringLength(20)>
    <JsonProperty(Order:=2, Required:=Required.Always)>
    Public Property Name As String

    <Required>
    <DataType(DataType.DateTime)>
    <JsonProperty(Order:=3)>
    Public Property Birth As DateTime

    <Required>
    <EnumDataType(GetType(BloodType))>
    <JsonProperty(Order:=4)>
    <JsonConverter(GetType(BloodTypeEnumConverter))>
    Public Property BloodType As BloodType

#End Region

#Region "Constructors"

    Public Sub New()
    End Sub

#End Region

#Region "Public Methods - Overrides"

    Public Overrides Function ToString() As String

        Return String.Format("ID        : {0}{4}" +
                             "Name      : {1}{4}" +
                             "Birth     : {2:yyyy/MM/dd}{4}" +
                             "BloodType : {3}{4}",
                             Id, Name, Birth, BloodType.ToString(), Environment.NewLine)
    End Function

#End Region

End Class

Public Enum BloodType
    Undefined
    A
    B
    O
    AB
End Enum

Public Class BloodTypeEnumConverter
    Inherits StringEnumConverter

    Public Overrides Function ReadJson(reader As JsonReader, objectType As Type, existingValue As Object, serializer As JsonSerializer) As Object

        Try
            Return MyBase.ReadJson(reader, objectType, existingValue, serializer)
        Catch ex As Exception
            Return BloodType.Undefined
        End Try

    End Function

End Class

File: JsonSchemaCSharp/Student.cs
using System;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;

namespace JsonSchemaCSharp {

    using Newtonsoft.Json;
    using Newtonsoft.Json.Converters;

    /// <summary>
    /// Student
    /// </summary>
    /// <remarks>
    /// For generating schema, and reading contents
    /// </remarks>
    public class Student {

        #region Properties

        [Required]
        [RegularExpression( @"^[A-Z][0-9]{7}?$" )]
        [JsonProperty( Order = 1, Required = Required.Always )]
        public string Id { get; set; }

        [Required]
        [StringLength( 20 )]
        [JsonProperty( Order = 2, Required = Required.Always )]
        public string Name { get; set; }

        [Required]
        [DataType( DataType.DateTime )]
        [JsonProperty( Order = 3 )]
        public DateTime Birth { get; set; }

        [Required]
        [EnumDataType( typeof( BloodType ) )]
        [JsonProperty( Order = 4 )]
        [JsonConverter( typeof( BloodTypeEnumConverter ) )]
        public BloodType BloodType { get; set; }

        #endregion

        #region Constructors

        public Student() { }

        #endregion

        #region Public Methods - Overrides

        public override string ToString() {

            return string.Format( @"ID        : {0}{4}" +
                                  @"Name      : {1}{4}" +
                                  @"Birth     : {2:yyyy/MM/dd}{4}" +
                                  @"BloodType : {3}{4}",
                                  Id, Name, Birth, BloodType.ToString(), Environment.NewLine );
        }

        #endregion
    }

    /// <summary>
    /// BloodType
    /// </summary>
    public enum BloodType {

        Undefined,
        A,
        B,
        O,
        AB,
    }

    /// <summary>
    /// StringEnumConverter for BloodType
    /// </summary>
    /// <remarks>
    /// If value is invalid on deserializing, to use default value.
    /// Overrides only ReadJson method.
    /// </remarks>
    public class BloodTypeEnumConverter : StringEnumConverter {

        public override object ReadJson(
            JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer ) {

            try {
                return base.ReadJson( reader, objectType, existingValue, serializer );
            }
            catch ( JsonSerializationException ) {
                return BloodType.Undefined;
            }
        }
    }
}
